docs/help/doc-code-samples.md

```diff
@@ -0,0 +1,51 @@
+# Custom code samples
+
+While we generate a basic `cURL` code sample by default in your documentation you might want to customize the example provided and share even more specific language code samples to your API consumers. This is why we added this custom property.
+
+Use the `x-codeSamples` property inside an Operation OpenAPI object. The `x-codeSamples` property accepts an array of code sample objects which are defined as such:
+
+| Property | Type   | Description                                                                                                  |
+|----------|--------|--------------------------------------------------------------------------------------------------------------|
+| lang *   | String | Code sample programming language                                                                             |
```

Comment:
are there limits or specific syntax for the lang parameter ? If yes, listing there here could be nice

Reply:
There are no hard limits. We use a basic “color syntax” on some commonly used languages (the ones supported by https://github.com/highlightjs/highlight.js/) but worst case scenario we wont have color syntax but the sample will still be displayed.

I don't have a strong opinion to add the language supported by our color syntaxing lib because for now the colors on the code samples bloc are not ideal (we only use **bold** to identify some keywords of languages, but don't actually use any colors in our custom color syntax CSS inside the code sample bloc) Maybe we can add the list when we improve the design of this code sample block?